app/controllers/hq/sponsorships_controller.rb

```diff
@@ -0,0 +1,37 @@
+class Hq::SponsorshipsController < HqController
+
+  def inactivate
+    sponsorship = Sponsorship.find(params[:id])
+
+    @sponsorship_inactivator = InactivateSponsorship.new(sponsorship: sponsorship,
+                                                         end_date: params[:sponsorship][:end_date])
+    inactivate_sponsorship and redirect_to hq_sponsor_path(sponsorship.sponsor_id)
+  end
+
+  def destroy
+    sponsorship = Sponsorship.find(params[:id])
+
+    @sponsorship_destructor = DestroySponsorship.new(sponsorship)
+
+    destroy_sponsorship and redirect_to hq_sponsor_path(sponsorship.sponsor_id)
+  end
```

Comment:
:+1: 
To complement this, it would be good to use [shallow nesting](http://guides.rubyonrails.org/routing.html#shallow-nesting) in routes.rb - collection actions remain scoped under the parent resource (here, sponsor), but member actions are not.

(Also, purely conceptually, destroying a sponsorship and then referencing it on L16 looks weird.)